stonesoup/dataassociator/neighbour.py

```diff
@@ -106,3 +111,92 @@ def associate(self, tracks, detections, time):
         associations = max(joint_hypotheses)
 
         return associations
+
+
+class GNNWith2DAssignment(DataAssociator):
+    """Global Nearest Neighbour Associator
+
+    Associates detections to a predicted state using the
+    Global Nearest Neighbour method, utilising a 2D matrix of
+    distances and a "shortest path" assignment algorithm.
+    """
+
+    hypothesiser = Property(
+        Hypothesiser,
+        doc="Generate a set of hypotheses for each prediction-detection pair")
+
+    def associate(self, tracks, detections, time):
+        """Associate a set of detections with predicted states.
+
+        Parameters
+        ----------
+        tracks : set of :class:`Track`
+            Current tracked objects
+        detections : set of :class:`Detection`
+            Retrieved measurements
+        time : datetime
+            Detection time to predict to
+
+        Returns
+        -------
+        dict
+            Key value pair of tracks with associated detection
+        """
+
+        # Convert sets to indexable lists
+        tracks = list(tracks)
+        detections = list(detections)
+
+        # Generate 2d array "matrix" of hypotheses mapping track to detection
+        hypothesis_matrix = np.empty(
+            [len(tracks), len(detections) + len(tracks)],
+            SingleDistanceHypothesis)
```

Comment:
Whilst this doesn't have an impact as _object_ array is created, this implies _distance_. Could just have `SingleHypothesis` or `object` to keep generic.